### ISSUE TITLE
`--bmc-depth` conflicts with `setUp` function

### ISSUE THREAD

**Issue by RaoulSchaffranek** [runtimeverification/evm-semantics#2011]
I got a runtime exception when generating an APRBMCProof for a contract with a `setup` function. I believe that we try to read the APRBMCProof as an APRProof here:

https://github.com/runtimeverification/evm-semantics/blob/5cda4c794acc592943f2c6322f11e0744884acc5/kevm-pyk/src/kevm_pyk/foundry.py#L964

The problem is that the `APRProof` parser expects the JSON to contain a property named `terminal_nodes`, but the `APRBMCProof` serializer does not generate such a field.

## Steps to reproduce:

~~~Solidity
// SPDX-License-Identifier: UNLICENSED
pragma solidity ^0.8.15;
contract MinimalSetup  {

    function setUp() external {
    }

    function doSomething() external {
    }

}
~~~

~~~bash
forge build
kevm foundry-compile --with-llvm-library
kevm foundry-prove --test MinimalSetup.doSomething --use-booster --bmc-depth 1
~~~

~~~
Traceback (most recent call last):
  File ".kevm-pyk-wrapped", line 9, in <module>
    sys.exit(main())
  File "kevm_pyk/__main__.py", line 87, in main
    execute(**vars(args))
  File "kevm_pyk/__main__.py", line 545, in exec_foundry_prove
    results = foundry_prove(
  File "kevm_pyk/foundry.py", line 602, in foundry_prove
    results = run_cfg_group(tests)
  File "kevm_pyk/foundry.py", line 590, in run_cfg_group
    _apr_proofs.append(_init_and_run_proof(init_problem))
  File "kevm_pyk/foundry.py", line 550, in _init_and_run_proof
    proof = _method_to_apr_proof(
  File "kevm_pyk/foundry.py", line 906, in _method_to_apr_proof
    kcfg, init_node_id, target_node_id = _method_to_cfg(
  File "kevm_pyk/foundry.py", line 947, in _method_to_cfg
    init_cterm = _init_cterm(
  File "kevm_pyk/foundry.py", line 1039, in _init_cterm
    accts, constraints = get_final_accounts_cell(init_state, kcfgs_dir)
  File "kevm_pyk/foundry.py", line 962, in get_final_accounts_cell
    apr_proof = APRProof.read_proof_data(proof_dir, proof_digest)
  File "pyk/proof/reachability.py", line 295, in read_proof_data
    terminal_nodes = proof_dict['terminal_nodes']
KeyError: 'terminal_nodes
~~~

**Comment by nwatson22** [comment]
I believe this is no longer an issue with the following:
```
forge build
kontrol build --with-llvm-library
kontrol prove --test MinimalSetup.doSomething --use-booster --bmc-depth 1
```
and that it was fixed by [pyk#601](https://github.com/runtimeverification/pyk/pull/601/files)

**Comment by palinatolmach** [comment]
Confirmed that the issue has been fixed.
